### PR TITLE
net: sockets: Implement non-blocking mode.

### PR DESCRIPTION
**Pull request by pfalcon**
Signed-off-by: Paul Sokolovsky <paul.sokolovsky@linaro.org>

### CODE REVIEW

File: subsys/net/lib/sockets/sockets.c [hunk 1]
```diff
@@ -290,3 +349,32 @@ ssize_t zsock_recv(int sock, void *buf, size_t max_len, int flags)
...
+
+/* As this is limited function, we don't follow POSIX signature, with
+ * "..." instead of last arg.
+ */
+int zsock_fcntl(int sock, int cmd, int flags)
+{
+	struct net_context *ctx = INT_TO_POINTER(sock);
+
+	switch (cmd) {
+
```
Comment (jukkar) [inline subsys/net/lib/sockets/sockets.c]: Empty line here looks bad, and we do not have this kind of style in switch() statements in other parts of the system.

File: subsys/net/lib/sockets/sockets.c [hunk 2]
```diff
@@ -290,3 +349,32 @@ ssize_t zsock_recv(int sock, void *buf, size_t max_len, int flags)
...
+	case F_GETFL:
+		if (sock_is_nonblock(ctx)) {
+		    return O_NONBLOCK;
+		}
+		return 0;
+
```
Comment (jukkar) [inline subsys/net/lib/sockets/sockets.c]: same here

File: subsys/net/lib/sockets/sockets.c [hunk 3]
```diff
@@ -290,3 +349,32 @@ ssize_t zsock_recv(int sock, void *buf, size_t max_len, int flags)
...
+	case F_SETFL:
+		if (flags & O_NONBLOCK) {
+			sock_set_flag(ctx, SOCK_NONBLOCK, SOCK_NONBLOCK);
+		} else {
+			sock_set_flag(ctx, SOCK_NONBLOCK, 0);
+		}
+		return 0;
+
```
Comment (jukkar) [inline subsys/net/lib/sockets/sockets.c]: and here too

**Review by jukkar — CHANGES_REQUESTED** [review]
Looks good, just very minor style issues.

File: subsys/net/lib/sockets/sockets.c (hunk 1 above)
Comment (pfalcon) [inline subsys/net/lib/sockets/sockets.c]: Sounds good, I just got to used that @tbursztyka asks to add more empty lines, not less ;-).

**Comment by pfalcon** [comment]
Fixed style issues.

**Review by jukkar — APPROVED** [review]

**Pull request merged**